crypto/crypto.go

```diff
@@ -247,6 +247,16 @@ func DecodePublicKeyHex(sigAlgo SignatureAlgorithm, s string) (PublicKey, error)
 	return DecodePublicKey(sigAlgo, b)
 }
 
+// It is possible in some cases where the resulting byte slice from the PEM
+// decoding of the X and Y points do not generate the right size byte slices
+// If we know there is an expected length for a specific signature algorithm,
+// we make sure to left pad it to the right length. This is a mapping of the
+// signature algorithms to the expected length of the byte slices representing
+// the X and Y points
+var expectedPointByteLength = map[SignatureAlgorithm]int{
+	crypto.ECDSAP256: 32,
+}
+
 // DecodePublicKeyHex decodes a PEM public key with the given signature algorithm.
 func DecodePublicKeyPEM(sigAlgo SignatureAlgorithm, s string) (PublicKey, error) {
 	block, _ := pem.Decode([]byte(s))
```

Comment:
I think we shouldn't have trailing data here.
```suggestion
	block, rest := pem.Decode([]byte(s))
	if len(rest)>0 {
	      return PublicKey{}, fmt.Errorf("crypto: failed to parse PEM string: %w", err)
	}
```